crates/autopilot/src/boundary/events/settlement.rs

```diff
@@ -41,7 +41,7 @@ impl EventStoring<contracts::gpv2_settlement::Event> for Indexer {
         OnSettlementEventUpdater::delete_observations(&mut transaction, from_block).await?;
         transaction.commit().await?;
 
-        self.settlement_updater.schedule_update();
```

Comment:
I don't remember off the top of my head (maybe github has some history), but was there a reason we scheduled this async?

Reply:
The PR that introduced this mentions:
>The goal of this PR is to bring the non-critical settlement updates into
the critical control flow, without making it itself blocking (ie the
critical code path just "schedules" execution of the non critical
updates).

While this seemed like the correct thing in the past since the `OnSettlementEventUpdater` logic now also has to be on the critical path I think it's fine to no longer have the `notify` mechanism here.
I guess technically we should continue to use the scheduling for the current mode and only do stuff blocking in the new mode. But given that the synchronized mode is our definitive goal this quarter and maintaining both modes of operations sounds like more work than it's worth I'd already switch it over to be completely blocking.

Reply:
Hm, this is not super reliable. 
For example, this [error](https://aws-es.cow.fi/_dashboards/goto/92724747850cbc1550969f565a702eac?security_tenant=global) causes <settlement, auction> relation not to be updated on time and waits for the next update (when next settlement is mined) to retry. 

Is your logic ok with this?